modules/tinymce/src/core/main/ts/newline/InsertDetailsNewLine.ts

```diff
@@ -0,0 +1,34 @@
+import Editor from '../api/Editor';
+import * as NewLineUtils from './NewLineUtils';
+
+export const isDetails = (node?: Node | null): node is HTMLDetailsElement =>
+  node?.nodeName === 'DETAILS';
+
+export const getDetailsRoot = (editor: Editor, element: Element): HTMLDetailsElement | null =>
+  editor.dom.getParent(element, isDetails);
+
+export const isLastEmptyBlockInDetails = (editor: Editor, shiftKey: boolean, element: Element): boolean => {
+  switch (true) {
```

Comment:
What's wrong with a simple boolean expression?

Reply:
It's not a style we use anywhere else. More readable for you may be a confusing mess for someone looking at this months or years from now. If you want to propose a new style, that's a team decision.

Injecting some personality is fine, but at a general level I try to encourage a similar coding style across the entire product for ease of maintenance. Every time you create a special unique snowflake it leaves a headache for someone to find later.